src/main/java/com/lpnu/shaggybeavers/service/impl/StorageServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.lpnu.shaggybeavers.service.impl;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.lpnu.shaggybeavers.service.StorageService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+
+@Service
+@RequiredArgsConstructor
+public class StorageServiceImpl implements StorageService {
+
+    @Value("")
+    private String bucketName;
+
+    private final AmazonS3 s3Client;
+
+    public URL uploadFile(MultipartFile multipartFile) throws IOException {
```

Comment:
you must handle exception there becuase if you do as you did there you will need to handle it in each method where that method calls